zjit.c

```diff
@@ -363,6 +360,12 @@ rb_zjit_defined_ivar(VALUE obj, ID id, VALUE pushval)
     return result ? pushval : Qnil;
 }
 
+bool
+rb_zjit_shape_too_complex_p(shape_id_t shape_id)
```

Comment:
Let's put this in jit.c as `rb_jit_shape_too_complex_p` and share it among YJIT and ZJIT. They're doing the same thing and we can avoid bloating the binary size for it.